server/lib/payments.js

```diff
@@ -39,16 +47,22 @@ export const executeOrder = (user, order, options) => {
   return order.populate()
     .then(() => {
       if (payment.interval) {
-        return models.Subscription.create(payment)
-          .then(subscription => {
-            order.subscription = subscription;
-            return order.update({ SubscriptionId: subscription.id });
-          })
+        return models.Subscription.create(payment).then(subscription => {
+          // The order instance doesn't have the Subscription field
+          // here because it was just created and no models were
+          // included so we're doing that manually here. Not the
+          // cutest but works.
+          order.Subscription = subscription;
+          libsubscription.updateNextChargeDate('success', order); // No DB access
```

Comment:
What will happen to `nextChargeDate` if the first payment (aka `processOrder`) fails?

Reply:
the `nextChargeDate` field will be updated to a month from the attempt date. but I moved the activation of the subscription from after this line to after executing `processOrder` successfully.